File: oop/code/oop/src/main/java/com/scaler/lld/bird/Bird.java
package com.scaler.lld.bird;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class Bird {
    private Integer weight;
    private String colour;
    private String size;
    private String beakType;
    private BirdType type;

    public void fly() {
        if (type == BirdType.Eagle) {
            System.out.println("Eagle is flying");
        } else if (type == BirdType.Penguin) {
            System.out.println("Penguin is swimming");
        } else if (type == BirdType.Parrot) {
            System.out.println("Parrot is flying");
        }
    }
}

File: python/oop/SOLID/bird/Bird.py
from abc import ABC, abstractmethod
import BirdType

class Bird(ABC):
	def __init__(self, weight, colour, size, beakType, birdType):
		self.weight = weight
		self.colour = colour
		self.size = size
		self.beakType = beakType
		self.birdType = birdType

	@abstractmethod
	def fly(self):
		pass